### PR TITLE
Add script support when making hot key authorisation certificates

### PR DESCRIPTION
**Pull request by carbolymer**
# Changelog

```yaml
- description: |
    Add plutus script support when making hot key authorisation certificates
# uncomment types applicable to the change:
  type:
   - feature        # introduces a new feature
   - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

A prerequisite for:
- https://github.com/IntersectMBO/cardano-cli/pull/641

# How to trust this PR
Trust no one.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Self-reviewed the diff

### CODE REVIEW

File: cardano-api/internal/Cardano/Api/Monad/Error.hs [hunk 1]
```diff
@@ -34,7 +35,7 @@ import           Data.Bifunctor (first)
 type MonadTransError e t m = (Monad m, MonadTrans t, MonadError e (t m))
 --
 -- | Same as 'MonadTransError', but with also 'MonadIO' constraint
-type MonadIOTransError e t m = (MonadIO m, MonadIO (t m), MonadTrans t, MonadError e (t m))
+type MonadIOTransError e t m = (MonadIO m, MonadIO (t m), MonadCatch m, MonadTrans t, MonadError e (t m))
```
Comment (smelc) [inline cardano-api/internal/Cardano/Api/Monad/Error.hs:38]: Not exactly sure why this is in this PR? It seems this type is used in `LedgerState.hs` so seems unrelated!?

Comment (carbolymer) [reply cardano-api/internal/Cardano/Api/Monad/Error.hs:38]: This type is use in CLI in: https://github.com/IntersectMBO/cardano-cli/pull/641

**Review by smelc — APPROVED** [review]
